includes/class-freemius.php

```diff
@@ -5644,14 +5644,14 @@ function _after_code_type_change() {
                 $this->_cache->expire( 'tabs_stylesheets' );
             }
 
-            if ( $this->is_registered() ) {
-                if ( ! $this->is_addon() ) {
-                    add_action(
-                        is_admin() ? 'admin_init' : 'init',
-                        array( &$this, '_plugin_code_type_changed' )
-                    );
-                }
+            if ( ! $this->is_addon() ) {
+                add_action(
+                    is_admin() ? 'admin_init' : 'init',
+                    array( &$this, '_plugin_code_type_changed' )
+                );
+            }
 
+            if ( $this->is_registered() ) {
                 if ( $this->is_premium() ) {

```

Comment:
@fajardoleo We can simplify `if ( $this->is_registered() && $this->is_premium() )`